code/game/objects/structures/razorwire.dm

```diff
@@ -0,0 +1,292 @@
+		M.visible_message("<span class='danger'>The barbed wire slices into [M]!</span>",
+		"<span class='danger'>The barbed wire slices into you!</span>", null, 5)
+		playsound(src, 'sound/effects/barbed_wire_movement.ogg', 25, 1)
+
+/obj/structure/razorwire/attack_alien(mob/living/carbon/Xenomorph/M)
+	M.animation_attack_on(src)
+	health -= rand(M.xeno_caste.melee_damage_lower, M.xeno_caste.melee_damage_upper)
+	playsound(src, 'sound/effects/barbed_wire_movement.ogg', 25, 1)
+	if(health <= 0)
+		M.visible_message("<span class='danger'>[M] slices [src] apart!</span>", \
+		"<span class='danger'>You slice [src] apart!</span>", null, 5)
+	else
+		M.visible_message("<span class='danger'>[M] slashes [src]!</span>", \
+		"<span class='danger'>You slash [src]!</span>", null, 5)
+	M.visible_message("<span class='danger'>The barbed wire slices into [M]!</span>",
+	"<span class='danger'>The barbed wire slices into you!</span>", null, 5)
+	M.apply_damage(rand(RAZORWIRE_BASE_DAMAGE * RAZORWIRE_MIN_DAMAGE_MULT_LOW, RAZORWIRE_BASE_DAMAGE * RAZORWIRE_MAX_DAMAGE_MULT_LOW)) //About a third as damaging as actually entering
+	update_health(TRUE)
+
+/obj/structure/razorwire/ex_act(severity)
+	switch(severity)
+		if(1.0)
+			visible_message("<span class='danger'>[src] is blown apart!</span>")
+			destroyed()
+			return
+		if(2.0)
+			health -= rand(33, 66)
+		if(3.0)
+			health -= rand(10, 33)
+	update_health()
+
+/obj/structure/razorwire/Bumped(atom/A)
+	. = ..()
+
+	if(istype(A, /mob/living/carbon/Xenomorph/Crusher))
+
+		var/mob/living/carbon/Xenomorph/Crusher/C = A
+
+		if(C.charge_speed < C.charge_speed_max * 0.5)
+			return
+
+		health -= 200 * round(C.charge_speed / max(1, C.charge_speed_max),0.01)
+		to_chat(world, "DEBUG: Crusher damage: [150 * round(C.charge_speed / max(1, C.charge_speed_max),0.01)]. Speed: [C.charge_speed] Max Speed: [C.charge_speed_max]")
+		update_health()
+
+		var/def_zone = ran_zone()
+		if(C.charge_speed >= C.charge_speed_max)
+			C.visible_message("<span class='danger'>[C] plows through the barbed wire!</span>",
+			"<span class='danger'>You plow through the barbed wire!</span>", null, 5)
+		else //If we didn't destroy the barbed wire, we get tangled up.
+			C.stop_momentum(C.charge_dir)
+			razorwire_tangle(C, RAZORWIRE_ENTANGLE_DELAY * 0.5) //entangled for only half as long
+
+		C.apply_damage(rand(RAZORWIRE_BASE_DAMAGE * RAZORWIRE_MIN_DAMAGE_MULT_MED, RAZORWIRE_BASE_DAMAGE * RAZORWIRE_MAX_DAMAGE_MULT_MED), BRUTE, def_zone, null, null, 1)
+		C.visible_message("<span class='danger'>The barbed wire slices into [C]!</span>",
+		"<span class='danger'>The barbed wire slices into you!</span>", null, 5)
+		playsound(src, 'sound/effects/barbed_wire_movement.ogg', 25, 1)
+
+
+
+/obj/structure/razorwire/bullet_act(obj/item/projectile/P)
+	bullet_ping(P)
+	health -= round(P.damage * 0.1)
+
+	if(istype(P.ammo, /datum/ammo/xeno/boiler_gas))

```

Comment:
Could open this exception to more xeno attacks, such as different spits and acid spray.